newlib/libc/machine/riscv/memcpy-asm.S

```diff
@@ -14,14 +14,14 @@
 .global memcpy
 .type	memcpy, @function
 memcpy:
-  mv t1, a0
+  mv a3, a0
   beqz a2, 2f
 
 1:
-  lb t2, 0(a1)
-  sb t2, 0(t1)
+  lbu a4, 0(a1)
+  sb a4, 0(a3)
   add   a2, a2, -1
```

Comment:
i have a preference to interleave the adds with the memory operation, in case, that there is some latency associated with the load operation, you reduce the possible stalls.